File: packages/core/src/schemas/epic.ts
import { z } from 'zod';
import {
  entityIdSchema,
  entityRefSchema,
  gitHubSyncSchema,
  gitLabSyncSchema,
  jiraSyncSchema,
  prioritySchema,
  statusSchema,
} from './common.js';

export const epicFrontmatterSchema = z.object({
  type: z.literal('epic'),
  id: entityIdSchema,
  title: z.string().min(1),
  status: statusSchema,
  priority: prioritySchema,
  owner: z.string().nullable().optional(),
  labels: z.array(z.string()).default([]),
  milestone_ref: entityRefSchema.nullable().optional(),
  github: gitHubSyncSchema.nullable().optional(),
  jira: jiraSyncSchema.nullable().optional(),
  gitlab: gitLabSyncSchema.nullable().optional(),
  created_at: z.string().optional(),
  updated_at: z.string().optional(),
});

export const epicSchema = epicFrontmatterSchema.extend({
  body: z.string().default(''),
  filePath: z.string(),
});

export type EpicFrontmatter = z.infer<typeof epicFrontmatterSchema>;
export type Epic = z.infer<typeof epicSchema>;

File: go/internal/schema/epic.go
package schema

import "time"

type EpicFrontmatter struct {
	Type         string      `yaml:"type" json:"type"`
	ID           string      `yaml:"id" json:"id"`
	Title        string      `yaml:"title" json:"title"`
	Status       Status      `yaml:"status" json:"status"`
	Priority     Priority    `yaml:"priority" json:"priority"`
	Owner        *string     `yaml:"owner,omitempty" json:"owner,omitempty"`
	Labels       []string    `yaml:"labels,omitempty" json:"labels,omitempty"`
	MilestoneRef *EntityRef  `yaml:"milestone_ref,omitempty" json:"milestone_ref,omitempty"`
	GitHub       *GitHubSync `yaml:"github,omitempty" json:"github,omitempty"`
	GitLab       *GitLabSync `yaml:"gitlab,omitempty" json:"gitlab,omitempty"`
	Jira         *JiraSync   `yaml:"jira,omitempty" json:"jira,omitempty"`
	CreatedAt    time.Time   `yaml:"created_at" json:"created_at"`
	UpdatedAt    time.Time   `yaml:"updated_at" json:"updated_at"`
}

type Epic struct {
	EpicFrontmatter `yaml:",inline"`
	Body            string `yaml:"-" json:"body"`
	FilePath        string `yaml:"-" json:"file_path"`
}
